pkg/operators/filter/filter.go

```diff
@@ -62,54 +62,50 @@ func (f *filterOperator) GlobalParams() api.Params {
 }
 
 func (f *filterOperator) InstanceParams() api.Params {
-	return api.Params{&api.Param{
-		Key: ParamFilter,
-		Description: `Filter rules
-  A filter can match any field using the following syntax:
-    field==value     - matches, if the content of field equals exactly value
-    field!=value     - matches, if the content of field does not equal exactly value
-    field>=value     - matches, if the content of field is greater than or equal to the value
-    field>value      - matches, if the content of field is greater than the value
-    field<=value     - matches, if the content of field is less than or equal to the value
-    field<value      - matches, if the content of field is less than the value
-    field~value      - matches, if the content of field matches the regular expression 'value'
-    field!~value     - matches, if the content of field does not match the regular expression 'value'
-                 see [https://github.com/google/re2/wiki/Syntax] for more information on the syntax
-  Multiple filters can be combined using a comma: field1==value1,field2==value2
-  It is recommended to use single quotes to escape the filter string, especially if using regular expressions.
-  Example: --filter 'field!~regex'
-        `,
-		Alias: "F",
-	}}
+	return api.Params{}
 }
 
-func keyForDataSource(ds datasource.DataSource) string {
-	return fmt.Sprintf("%s.%s", ParamFilterExpr, ds.Name())
+func keyForDataSource(param string, ds datasource.DataSource) string {
+	return fmt.Sprintf("%s.%s", param, ds.Name())
 }
 
 func (f *filterOperator) InstantiateDataOperator(gadgetCtx operators.GadgetContext, instanceParamValues api.ParamValues) (operators.DataOperatorInstance, error) {
-	filterCfg := instanceParamValues[ParamFilter]
-
 	fop := &filterOperatorInstance{
 		gadgetCtx: gadgetCtx,
 		ffns:      map[datasource.DataSource][]func(datasource.DataSource, datasource.Data) bool{},
 	}
 
-	filters := api.SplitStringWithEscape(filterCfg, ',')
-	for _, filter := range filters {
-		if filter == "" {
-			continue
+	datasources := gadgetCtx.GetDataSources()
+	for _, ds := range datasources {
+		key := keyForDataSource(ParamFilter, ds)
+		if filterStr := instanceParamValues[key]; filterStr != "" {
+			filters := api.SplitStringWithEscape(filterStr, ',')
+			for _, filter := range filters {
+				if filter == "" {
+					continue
+				}
+				gadgetCtx.Logger().Debugf("adding filter %q", filter)
+				err := fop.addFilter(gadgetCtx, ds, filter)
+				if err != nil {
+					return nil, err
+				}
+			}
 		}
```

Comment:
What about creating a fop.addFilters method to avoid duplicating this code few lines below?